### PR TITLE
SE-1463 Fix *_ENV_EXTRA configuration

### PR DESCRIPTION
**Pull request by samuelallan72**
This PR re-enables support for `EDXAPP_{CMS,LMS}_ENV_EXTRA` snippets.
Previously these extra variables were only merged here:

https://github.com/edx/configuration/blob/master/playbooks/roles/edxapp/templates/lms.env.json.j2#L1

Now that edx-platform doesn't read from the json config files, this behaviour
must be ported to work with the new yml config.

**JIRA tickets**: [OSPR-3838](https://openedx.atlassian.net/browse/OSPR-3838)

**Dependencies**: None

**Sandbox URL**:

- lms: https://se-1071.sandbox.opencraft.hosting/

(This sandbox is for another ticket, but was deployed from this
configuration branch with the settings as at the end of this PR description.)

**Merge deadline**: None

**Testing instructions**:

1. add custom config to EDXAPP_LMS_ENV_EXTRA and EDXAPP_CMS_ENV_EXTRA
2. deploy an instance of edx platform using this configuration branch
3. check that the config added under the above variables are visible as top level
   keys in lms.yml and studio.yml respectively.

For example, if you add to the extra configuration:

```
EDXAPP_CMS_ENV_EXTRA:
  DUMMY: true
```

You should see this line in studio.yml:

```
DUMMY: true
```

(Previously, this would not end up with studio.yml; would only be in
cms.env.json)

**Author notes and concerns**:

**Reviewers**
- [x] @viadanna
- [x] edX reviewer[s] TBD

**Settings**
```yaml
EDXAPP_LMS_ENV_EXTRA:
  ADDL_INSTALLED_APPS:
   - openedx.core.djangoapps.heartbeat
  HEARTBEAT_EXTENDED_CHECKS:
   - "openedx.core.djangoapps.django_comment_common.comment_client.utils.check_forum_heartbeat"
   - "openedx.core.djangoapps.heartbeat.default_checks.check_celery"
```

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @swalladge! I've created [OSPR-3838](https://openedx.atlassian.net/browse/OSPR-3838) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by natabene** [comment]
@swalladge Thank you for your contribution. Please let me know once this is ready for our review.

File: playbooks/roles/nltk/defaults/main.yml [hunk 1]
```diff
@@ -11,3 +11,5 @@ NLTK_DATA:
       url: "http://nltk.github.io/nltk_data/packages/corpora/stopwords.zip" }
   - { path: "corpora/wordnet",
       url: "http://nltk.github.io/nltk_data/packages/corpora/wordnet.zip" }
+
+NLTK_DOWNLOAD_TIMEOUT: 100
```
Comment (viadanna) [inline playbooks/roles/nltk/defaults/main.yml]: Will this unrelated change be removed prior to merging?

**Review by viadanna — APPROVED** [review]
Good to go 👍

Just a question on the unrelated commit.

- [x] I tested this on a sandbox, by provisioning it with `LMS_ENV_EXTRA` and `CMS_ENV_EXTRA` extra settings and they were added correctly to the lms.yml and studio.yml
- [x] I read through the code.
- [ ] ~I checked for accessibility issues.~
- [ ] ~Includes documentation.~

File: playbooks/roles/nltk/defaults/main.yml (hunk 1 above)
Comment (samuelallan72) [reply playbooks/roles/nltk/defaults/main.yml]: Ah yes, this was added because of how I was testing it. I will remove that commit and force push now.

Comment (samuelallan72) [reply playbooks/roles/nltk/defaults/main.yml]: done :)

**Comment by samuelallan72** [comment]
@natabene This is ready for edX review, thanks. :)

**Comment by natabene** [comment]
@edx/devops This is ready for your review.

**Comment by samuelallan72** [comment]
@natabene Any updates on when this might be reviewed? It is high priority for us, as currently `EDXAPP_{CMS,LMS}_ENV_EXTRA` support is broken on master and it affects all master sandboxes.

**Comment by natabene** [comment]
@swalladge No updates, I will see what we can do this week, though.

**Comment by natabene** [comment]
@fredsmith Can you give this a look please?

**Review by fredsmith — APPROVED** [review]
:shipit:

**Comment by natabene** [comment]
@fredsmith Thank you, is this ready to merge? If so, can you merge?

**Comment by openedx-webhooks** [comment]
@swalladge 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/configuration&entry.867055334=https://github.com/edx/configuration/pull/5400&entry.1484655318=https://github.com/swalladge&entry.752974735=2019-09-10+03:28&entry.1917517419=2019-10-02+16:01&entry.2133058324=Yes) so we can improve your experience in the future.
